scipy/stats/_continuous_distns.py

```diff
@@ -3891,34 +3917,53 @@ def f(theta):
             elif x0 == zeta:
                 return sc.gamma(1+1/alpha)*np.cos(xi)/np.pi/((1+zeta**2)**(1/alpha/2))
             else:
-                return levy_stable_gen._pdf_single_value_zolotarev(-x, alpha, -beta)
+                return levy_stable_gen._pdf_single_value_piecewise(-x, alpha, -beta)
         else:
             # since location zero, no need to reposition x for S_0 parameterization
             xi = np.pi/2
             if beta != 0:
-                warnings.warn('Density calculation unstable for alpha=1 and beta!=0.' +
-                              ' Use quadrature method instead.', RuntimeWarning)
-
                 def V(theta):
                     expr_1 = np.pi/2+beta*theta
-                    return 2. * expr_1 * np.exp(expr_1*np.tan(theta)/beta) / np.cos(theta) / np.pi
+                    return 2. * expr_1 * np.exp(expr_1*np.tan(theta)/beta) / np.cos(theta) / np.pi 
 
                 def g(theta):
                     return np.exp(-np.pi * x / 2. / beta) * V(theta)
 
                 def f(theta):
-                    return g(theta) * np.exp(-g(theta))
+                    g_1 = g(theta)
+                    # as x*exp(-x) -> 0 as x -> inf
+                    return 0 if g_1 == np.inf else g_1 * np.exp(-g_1)
 
                 with np.errstate(all="ignore"):
-                    intg_max = optimize.minimize_scalar(lambda theta: -f(theta), bounds=[-np.pi/2, np.pi/2])
-                    intg = integrate.fixed_quad(f, -np.pi/2, intg_max.x)[0] + integrate.fixed_quad(f, intg_max.x, np.pi/2)[0]
+                    peak = optimize.ridder(lambda t2: g(t2) - 1, -np.pi/2, np.pi/2)
+
+                    # find better upper bound as
+                    # quadpack adaptive quadrature
+                    # unable to handle long zero tails.
+                    # TODO: better approach?
+                    upper_interval = np.pi/2-peak
+                    upper_bound = np.pi/2
+                    delta_sign = 1
+                    test_point = peak
+                    for i in range(20):
+                        delta = upper_interval/2.**i
+                        test_point = test_point + delta_sign*delta
+                        if np.isclose(f(test_point), 0, rtol=1e-013, atol=1e-013):
+                            upper_bound = test_point
+                            delta_sign = -1
+                        else:
+                            delta_sign = 1
+                
+                    intg = integrate.quad(f, -np.pi/2, peak)[0]\
```

Comment:
I think we try to avoid these trailing `\` in SciPy code -- probably better to use parentheses to achieve the line continuation

Reply:
> It's always nice to see proposals to improve scipy, and if I had infinite time I'd dig into this, but I don't, so I won't be able to do a real review of this in the short term. Instead, I'll just be _that guy_ and nag about [PEP-8](https://www.python.org/dev/peps/pep-0008/). 😝 Please keep all lines less than 80 characters.

Why doesn't the automated checker pick up the PEP-8 stuff? That's what I went by with my original PR that introduced all this code in the 1st place. Perhaps the next person with _infinite time_ could see what's happening there (pycodestyle etc)? Obviously, I'll try to fix the line lengths now I know some potential reviewers are still stuck with small VDUs :stuck_out_tongue_winking_eye:

Reply:
There is a lot of older code in scipy that does not conform to PEP-8, so we don't have all the possible pycodestyle checks enabled.  We're trying to ensure that *new* code follows the style guidelines.  It is expected that reviewers will keep an eye on PEP-8 compliance as part of the review process.

When I work on bigger monitors, I usually take advantage of the screen space by editing with two or more full-width tabs side-by-side, so the 79 character limit is still a useful convention--for me, anyway.

(And ok, yeah,  I wouldn't actually need *infinite* time 😧, but just a lot more than I'll have in the next few weeks.)

Reply:
This file is bothering my OCD for a long-time already but I didn't touch it since a lot of open PRs will have clashes if I fix all of them 😃

Reply:
What I'm finding is by updating the PR to use 79 char limit, it's is heavily obfuscating what code is actually changing functionality. I feel it would be better if I made the line length changes on a separate PR.

Reply:
Changed PR to enforce 79 char limit and PEP8 spacing/line continuations as much as possible. 

I see all checks pass except a pypy check that appears to be a test harness setup issue rather than something to do with PR.

Reply:
Ok, we can ignore the pypy CI stuff for now -- it has been removed from master branch for the time being anyway